File: LeetCode/2161/Solution.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> ret(nums.size(), 0);
        
        int small = 0, eq = 0;
        
        for(int n : nums) {
            if(pivot > n)
                small++;
            else if(pivot == n)
                eq++;
        }
        
        int index=0;
        int eqBase = small;
        int bigBase = small+eq;
        for(int n : nums) {
            if(n < pivot) {
                ret[index++] = n;
            } else if(n == pivot) {
                ret[eqBase++] = n;
            } else {
                ret[bigBase++] = n;
            }
        }
        
        return ret;
    }
};

File: LeetCode/2161/Solution.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] result = new int[n];
        int index = 0;
        for(int i=0;i<n;++i) {
            if(nums[i] < pivot) {
                result[index++] = nums[i];
            }
        }
        for(int i=0;i<n;++i) {
            if(nums[i] == pivot) {
                result[index++] = nums[i];
            }
        }
        for(int i=0;i<n;++i) {
            if(nums[i] > pivot) {
                result[index++] = nums[i];
            }
        }
        return result;
    }
}
